app/src/main/java/org/mozilla/focus/utils/IntentUtils.kt

```diff
@@ -3,142 +3,68 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 package org.mozilla.focus.utils
 
-import android.content.ActivityNotFoundException
+import android.content.ComponentName
 import android.content.Context
 import android.content.Intent
-import android.net.Uri
-import androidx.annotation.StringRes
-import androidx.appcompat.app.AlertDialog
-import org.mozilla.focus.R
-import java.net.URISyntaxException
+import android.content.pm.LabeledIntent
+import android.os.Build
+import android.os.Parcelable
 
 object IntentUtils {
-    private const val MARKET_INTENT_URI_PACKAGE_PREFIX = "market://details?id="
-    private const val EXTRA_BROWSER_FALLBACK_URL = "browser_fallback_url"
 
-    /**
-     * Find and open the appropriate app for a given Uri. If appropriate, let the user select between
-     * multiple supported apps. Returns a boolean indicating whether the URL was handled. A fallback
-     * URL will be opened in the supplied WebView if appropriate (in which case the URL was handled,
-     * and true will also be returned). If not handled, we should  fall back to webviews error handling
-     * (which ends up calling our error handling code as needed).
-     *
-     * Note: this method "leaks" the target Uri to Android before asking the user whether they
-     * want to use an external app to open the uri. Ultimately the OS can spy on anything we're
-     * doing in the app, so this isn't an actual "bug".
-     */
-    fun handleExternalUri(context: Context, uri: String): Boolean {
-        // This code is largely based on Fennec's ExternalIntentDuringPrivateBrowsingPromptFragment.java
-        val intent = try { Intent.parseUri(uri, 0) } catch (e: URISyntaxException) { return false }
-
-        // Since we're a browser:
-        intent.addCategory(Intent.CATEGORY_BROWSABLE)
-
-        // This is where we "leak" the uri to the OS. If we're using the system webview, then the OS
-        // already knows that we're opening this uri. Even if we're using GeckoView, the OS can still
-        // see what domains we're visiting, so there's no loss of privacy here:
-        val packageManager = context.packageManager
-        val matchingActivities = packageManager.queryIntentActivities(intent, 0)
-
-        when (matchingActivities.size) {
-            0 -> handleUnsupportedLink(context, intent)
-            1 -> {
-                val info = matchingActivities[0]
-                val externalAppTitle = info?.loadLabel(packageManager) ?: "(null)"
+    fun getIntentChooser(
+        context: Context,
+        intent: Intent,
+        chooserTitle: CharSequence? = null
+    ): Intent? {
+        val chooserIntent: Intent?
```

Comment:
Am I correct in saying that this will never be null?
So we can declare this as non-nullable, maybe as a `lateinit var` and then the code would be a bit more straightforward in clients also - no need to check the Intent for null before `startActivity()`
Otherwise we might have to think about how to handle cases in which this is null <=> users tap to share but nothing happens.